toolchain/lex/lexer.h

```diff
@@ -55,10 +55,7 @@ class Lexer {
         token_translator_(&buffer_),
         token_emitter_(token_translator_, consumer_) {}
 
-  // Find all line endings and create the line data structures. Explicitly kept
-  // out-of-line because this is a significant loop that is useful to have in
-  // the profile and it doesn't simplify by inlining at all. But because it can,
-  // the compiler will flatten this otherwise.
```

Comment:
It seems like I was missing the actual annotation, but I think this comment is still accurate and we should add an annotation here?

The over-eager inlining of this makes the profile less helpful.

Reply:
TBH here I'd thought that the refactoring would stop inlining due to the .h split. Restored.